src/Annotation/FieldEncryptionProvider.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\field_encrypt\Annotation\FieldEncryptionProvider.
+ */
+
+namespace Drupal\field_encrypt\Annotation;
+
+use Drupal\Component\Annotation\Plugin;
+
+/**
+ * Defines a FieldEncryptionProvider annotation object.
+ *
+ * @Annotation
+ */
+class FieldEncryptionProvider extends Plugin {
+  /**
+   * The machine name for the plugin.
+   */
+  public $id;
+
+  /**
+   * The human-readable name for the plugin.
+   *
+   * @ingroup plugin_translatable
+   *
+   * @var \Drupal\Core\Annotation\Translation
+   */
+  public $label;
+
+  /**
+   * Defines a mapping of field types and their properties to be encrypted.
```

Comment:
I am still wondering if this limits flexibility.

I think having properties be selected from within the interface is ideal instead of a plugin system. That basically means other contrib field types can leverage it automatically. 

For use cases like file_encrypt, there are entirely separate hooks for loading/storing files. These hooks, completely unique to the field metadata and field "values" stored in the database need to be explored. But, I think it's out of the scope of field_encrypt. We can certainly evaluate file_encrypt as well (e.g. any metadata or aspects of a file type field stored in the db), but the actual encryption of the file itself is out of scope.